internal/app/machined/pkg/runtime/v1alpha1/platform/nocloud/metadata.go

```diff
@@ -253,6 +254,21 @@ func (n *Nocloud) acquireConfig(ctx context.Context, r state.State) (metadataCon
 		_ = yaml.Unmarshal(metadataConfigDl, metadata) //nolint:errcheck
 	}
 
+	// Some providers may provide the hostname via user-data instead of meta-data (e.g. Proxmox VE)
+	// As long as the user doesn't use it for machine config, it can still be used to obtain the hostname
+	if machineConfigDl != nil {
+		fallbackMetadata := &MetadataConfig{}
+		_ = yaml.Unmarshal(machineConfigDl, fallbackMetadata) //nolint:errcheck
+
```

Comment:
What do you think to check `metadata.Hostname` fist to not Unmarshal machineConfigDl.
Or check header `#cloud-config` in machineConfigDl before Unmarshal?

yaml.Unmarshal is cpu expensive function. And in most of the cases machineConfigDl will non empty array.

All other changes is good for me 👍

Reply:
> What do you think to check metadata.Hostname fist to not Unmarshal machineConfigDl.
> yaml.Unmarshal is cpu expensive function. And in most of the cases machineConfigDl will non empty array.

OK, that makes sense

> Or check header #cloud-config in machineConfigDl before Unmarshal?

Currently NoCloud platform in Talos have no support for anything else:

https://github.com/siderolabs/talos/blob/9c48824f7607702ee96f82112b5132aaf7a5eaa4/internal/app/machined/pkg/runtime/v1alpha1/platform/nocloud/nocloud.go#L86-L88

Reply:
Fixed